lasagne/regularization.py

```diff
@@ -1,15 +1,131 @@
 import theano.tensor as T
+from .layers import Layer, get_all_params
 
-from . import layers
 
+def l1(x):
+    """Computes the L1 norm of a tensor
 
-def l2(layer, include_biases=False):
-    if include_biases:
-        all_params = layers.get_all_params(layer)
-    else:
-        all_params = layers.get_all_non_bias_params(layer)
+    Parameters
+    ----------
+    x : Theano tensor
 
-    return sum(T.sum(p**2) for p in all_params)
+    Returns
+    -------
+    Theano scalar
+        l1 norm (sum of absolute values of elements)
+    """
+    return T.sum(abs(x))
 
 
-# TODO: sparsity regularization
+def l2(x):
+    """Computes the squared L2 norm of a tensor
+
+    Parameters
+    ----------
+    x : Theano tensor
+
+    Returns
+    -------
+    Theano scalar
+        squared l2 norm (sum of squared values of elements)
+    """
+    return T.sum(x**2)
+
+
+def apply_penalty(tensor_or_tensors, penalty, **kwargs):
+    """
+    Computes the total cost for applying a specified penalty
+    to a tensor or group of tensors.
+
+    Parameters
+    ----------
+    tensor_or_tensors : Theano tensor or list of tensors
+    penalty : callable
+    kwargs : keyword arguments passed to penalty
+
+    Returns
+    -------
+    Theano scalar
+        a scalar expression for the total penalty cost
+    """
+    try:
+        return sum(penalty(x, **kwargs) for x in tensor_or_tensors)
+    except (TypeError, ValueError):
+        return penalty(tensor_or_tensors, **kwargs)
+
+
+def regularize_layer_params(layer, penalty,
+                            tags={'regularizable': True}, **kwargs):
+    """
+    Computes a regularization cost by applying a penalty to the parameters
+    of a layer or group of layers.
+
+    Parameters
+    ----------
+    layer : a :class:`Layer` instances or list of layers.
+    penalty : callable
+    tags: dict of tags used to filter the parameters of the layer or layers.
+        By default, only parameters with the `regularizable` tag are included.
+    kwargs : keyword arguments passed to penalty.
```

Comment:
Same here. It should be:
```python
tags : dict
    Tag specification to filter the parameters of the layer or layers.
    By default, ...
```
For `kwargs`, I'm not sure how to document it, see above. In `get_params()`, they are documented as `**tags`, and in `get_output_for()`, they are not documented at all.